### PR TITLE
fix(ActivityCenter): Fixing showing AC notifications

### LINKED ISSUE #7016: Activity center - Activity not being shown
# Bug Report

Epic: https://github.com/status-im/status-desktop/issues/7070

## Description

Found during testing https://github.com/status-im/status-desktop/pull/6508#issuecomment-1213051052 (item 10)

It's not currently possible to populate the activity center with activity.

## Steps to reproduce

Perform activity that would populate activity center, such as a mention.
Find that no notification is generated for the relevant user.

#### Expected behavior

User receives notification in Activity Center.

#### Actual behavior

No notifications in Activity Center

### Additional Information

- Status desktop version: Master, last checked 12/Aug and https://github.com/status-im/status-desktop/pull/6508
- Operating System: Mac

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #7016

Getting any of AC notifications visible

### Affected areas

Activity Center

### Screenshot of functionality (including design for comparison)

Screenshot 2022-09-20 at 10 20 02

### CODE REVIEW

File: ui/app/mainui/activitycenter/views/ActivityNotificationBase.qml [hunk 1]
```diff
@@ -21,10 +21,6 @@ Item {
 
     StatusFlatRoundButton {
         id: markReadBtn
-        width: 32
-        height: 32
-        icon.width: 24
-        icon.height: 24
         anchors.verticalCenter: parent.verticalCenter
```
Comment (caybro) [inline ui/app/mainui/activitycenter/views/ActivityNotificationBase.qml:24]: there's no `parent` around

File: src/app_service/service/activity_center/service.nim [hunk 2]
```diff
@@ -103,20 +103,13 @@ QtObject:
...
-    if(self.cursor == ""): return
-
     var cursorVal: JsonNode
 
-    if self.cursor == "":
-      cursorVal = newJNull()
-    else:
-      cursorVal = newJString(self.cursor)
-
+    cursorVal = newJString(self.cursor)
```
Comment (caybro) [inline src/app_service/service/activity_center/service.nim:108]: Initialize directly in the declaration?

**Review by caybro — APPROVED** [review]
Looks good, just some remarks

File: ui/app/mainui/activitycenter/views/ActivityNotificationBase.qml (hunk 1 above)
Comment (MishkaRogachev) [reply ui/app/mainui/activitycenter/views/ActivityNotificationBase.qml:24]: Root item is parent for StatusFlatRoundButton

File: src/app_service/service/activity_center/service.nim (hunk 2 above)
Comment (MishkaRogachev) [reply src/app_service/service/activity_center/service.nim:108]: turned back

File: ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    height: Math.max(60, notificationMessage.height)
+
+    MessageView {
+        id: notificationMessage
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.right: parent.right
+        z: -1
+
+        rootStore: root.store
```
Comment (caybro) [inline ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml:28]: There is no `root.store` property, how does this work?

File: ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        messageStore: root.store.messageStore
+        messageContextMenu: root.messageContextMenu
+
+        messageId: notification.id
```
Comment (caybro) [inline ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml:32]: Same for `notification`

File: src/app_service/service/activity_center/service.nim [hunk 5]
```diff
@@ -115,8 +113,7 @@ QtObject:
     let callResult = backend.activityCenterNotifications(cursorVal, DEFAULT_LIMIT)
     let activityCenterNotificationsTuple = parseActivityCenterNotifications(callResult.result)
 
-    self.cursor = activityCenterNotificationsTuple[0];
-
+    # self.cursor = activityCenterNotificationsTuple[0];
```
Comment (caybro) [inline src/app_service/service/activity_center/service.nim:116]: Why commented out?

**Review by caybro — CHANGES_REQUESTED** [review]

**Review by saledjenic — APPROVED** [review]

File: ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml (hunk 3 above)
Comment (MishkaRogachev) [reply ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml:28]: ActivityNotificationMessage is derived from ActivityNotificationBase which has store and notification

File: src/app_service/service/activity_center/service.nim (hunk 5 above)
Comment (MishkaRogachev) [reply src/app_service/service/activity_center/service.nim:116]: Temporary, my second issue is about cursor fixes. Turned it back for this PR

File: ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml (hunk 3 above)
Comment (caybro) [reply ui/app/mainui/activitycenter/views/ActivityNotificationMessage.qml:28]: Ah! :)

**Review by caybro — APPROVED** [review]

**Pull request merged**